internal/compiler/generator/rust.rs

```diff
@@ -791,6 +788,22 @@ fn generate_sub_component(
         }
     }
 
+    let mut item_geometry_branch = component
+        .geometries
+        .iter()
+        .enumerate()
+        .filter_map(|(i, x)| x.as_ref().map(|x| (i, x)))
+        .map(|(index, expr)| {
+            let expr = compile_expression(&expr.borrow(), &ctx);
+            let index = index as u32;
+            quote!(#index => {
+                // The result of the expression is an anonymous struct, so fields are in alphabetical order
+                let (h, w, x, y) = #expr;
+                sp::euclid::rect(x, y, w, h)
```

Comment:
Ahh, this is what you meant the code size increment. But why do that here instead of in line 1037?